src/libbson/tests/test-bson-vector.c

```diff
@@ -0,0 +1,1581 @@
+      memset (bson_reserve_buffer (&doc, doc.len + reserve_len) + doc.len, 0xdd, reserve_len);
+
+      bson_vector_packed_bit_view_t view;
+      ASSERT (BSON_APPEND_VECTOR_PACKED_BIT_UNINIT (&doc, "padding_init_test", 12, &view));
+      ASSERT (bson_vector_packed_bit_view_length_bytes (view) == 2);
+      ASSERT (bson_vector_packed_bit_view_padding (view) == 4);
+
+      // BSON validity only requires the low 4 bits to be zero, but the entire last
+      // byte will be zeroed by our implementation.
+      uint8_t bytes[2];
+      ASSERT (bson_vector_packed_bit_view_read_packed (view, bytes, sizeof bytes, 0));
+      ASSERT_CMPUINT ((unsigned) bytes[0], ==, 0xdd);
+      ASSERT_CMPUINT ((unsigned) bytes[1], ==, 0x00);
+   }
+
+   // Padding bits can't be forcibly given nonzero values using bson_vector_packed_bit_view_write_packed
+   {
+      bson_vector_packed_bit_view_t view;
+      ASSERT (BSON_APPEND_VECTOR_PACKED_BIT_UNINIT (&doc, "padding_mask_test", 13, &view));
+      ASSERT (bson_vector_packed_bit_view_length_bytes (view) == 2);
+      ASSERT (bson_vector_packed_bit_view_padding (view) == 3);
+
+      uint8_t bytes[2] = {0xff, 0xff};
+      ASSERT (bson_vector_packed_bit_view_write_packed (view, bytes, sizeof bytes, 0));
+      ASSERT (bson_vector_packed_bit_view_read_packed (view, bytes, sizeof bytes, 0));
+      ASSERT_CMPUINT ((unsigned) bytes[0], ==, 0xff);
+      ASSERT_CMPUINT ((unsigned) bytes[1], ==, 0xf8);
+   }
+
+   bson_destroy (&doc);
+}
+
+// Note: The effective MAX_TESTED_VECTOR_LENGTH is limited to 2047 on Windows due to RAND_MAX==0x7fff
+#define MAX_TESTED_VECTOR_LENGTH 10000
+#define FUZZ_TEST_ITERS 5000
+
+static void
+test_bson_vector_view_api_fuzz_int8 (void)
+{
+   size_t current_length = 0;
+   bson_t vector_doc = BSON_INITIALIZER;
+   int8_t *expected_elements = bson_malloc (MAX_TESTED_VECTOR_LENGTH * sizeof *expected_elements);
+   int8_t *actual_elements = bson_malloc (MAX_TESTED_VECTOR_LENGTH * sizeof *actual_elements);
+   for (int fuzz_iter = 0; fuzz_iter < FUZZ_TEST_ITERS; fuzz_iter++) {
+      unsigned r = (unsigned) rand ();
+      unsigned r_operation = r & 0xFu;
+      size_t r_param = r >> 4;
+
+      if (current_length == 0 || r_operation == 15) {
+         // Resize and fill
+         size_t new_length = (size_t) r_param % MAX_TESTED_VECTOR_LENGTH;
+         bson_reinit (&vector_doc);
+         bson_vector_int8_view_t view;
+         BSON_ASSERT (BSON_APPEND_VECTOR_INT8_UNINIT (&vector_doc, "vector", new_length, &view));
+         for (size_t i = 0; i < new_length; i++) {
+            expected_elements[i] = (int8_t) (uint8_t) rand ();
+         }
+         BSON_ASSERT (bson_vector_int8_view_write (view, expected_elements, new_length, 0));
+         current_length = new_length;
+
+      } else if (r_operation < 7) {
+         // Partial write
+         size_t element_count = r_param % current_length;
+         size_t offset = rand () % (current_length - element_count);
```

Comment:
```suggestion
         size_t offset = (size_t) rand () % (current_length - element_count);
```

Explicit invocation of `rand ()` in modulo expressions still require an explicit cast to address `-Wsign-conversion` warnings for `int` -> `size_t`.